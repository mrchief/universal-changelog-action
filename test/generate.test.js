const { generate } = require("../lib/generate")
const commits = require("./fixtures/commits.json")
const comparison = require("./fixtures/comparison.json")

test("generates changelog from commits", async () => {
  const previousReleaseTagNameOrSha = "2020-07-26T191652Z"
  const nextReleaseTagName = "2020-08-02T034148Z"
  const nextReleaseName = "Release 2020-08-02T034148Z"

  const changelog = await generate({
    nextReleaseTagName,
    previousReleaseTagNameOrSha,
    commits,
    nextReleaseName,
  })

  expect(changelog).toMatchSnapshot()
})

test("generates changelog from refs", async () => {
  const previousReleaseTagNameOrSha = "2020-07-26T191652Z"
  const nextReleaseTagName = "2020-08-02T034148Z"
  const nextReleaseName = "Release 2020-08-02T034148Z"

  const changelog = await generate({
    nextReleaseTagName,
    previousReleaseTagNameOrSha,
    commits: comparison.commits,
    nextReleaseName,
  })

  expect(changelog).toMatchSnapshot()
})
