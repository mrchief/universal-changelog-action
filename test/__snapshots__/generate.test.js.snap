// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates changelog from commits 1`] = `
"## [Release 2020-08-02T034148Z](https://github.com/owner/repo/releases/tag/2020-08-02T034148Z)


### ‚ú® Features

- If we parse the application, we can get to the JSON panel through the 1080p JSON array! ([\`50b1c2a\`](https://github.com/acme/foo/commit/50b1c2ae17634194843ca38b0b69993e))
- We need to navigate the haptic SSL program! ([\`b9af129\`](https://github.com/acme/foo/commit/b9af129654524dc7bb7ed8d09684baf8))

### üêõ Fixes

- Use the 1080p HTTP card, then you can input the optical program! ([\`484edcb\`](https://github.com/acme/foo/commit/484edcb1e9154eb3bf861edd35d96c4b))



### üõ† Maintenance

- We need to index the bluetooth PNG pixel! ([\`048bd17\`](https://github.com/acme/foo/commit/048bd17682d84af6a061f2d61c418fde))

### ‚úÖ Tests

- I'll override the digital CSS capacitor, that should bus the EXE alarm! ([\`b04b7d5\`](https://github.com/acme/foo/commit/b04b7d5827d240768b6d2126732cef41))
- You can't generate the system without hacking the multi-byte HDD panel! ([\`d62ac61\`](https://github.com/acme/foo/commit/d62ac612cc36411f9091ee673b9fb265))
- Try to override the COM protocol, maybe it will calculate the solid state capacitor! ([\`0e72523\`](https://github.com/acme/foo/commit/0e72523714e1498dac5639edd805a050))

### üìö Documentation

- navigating the interface won't do anything, we need to transmit the digital THX card! ([\`59ce0fc\`](https://github.com/acme/foo/commit/59ce0fcbdefb4b8487319126bb877a04))
- We need to generate the bluetooth JBOD matrix! ([\`da66679\`](https://github.com/acme/foo/commit/da666790429b4757b60bcd2eb4d05435))
- We need to back up the online PCI interface! ([\`e836705\`](https://github.com/acme/foo/commit/e836705301dc4e93a3c515e5d083d731))

"
`;

exports[`generates changelog from refs 1`] = `
"## [Release 2020-08-02T034148Z](https://github.com/owner/repo/releases/tag/2020-08-02T034148Z)


### ‚ú® Features

- If we parse the application, we can get to the JSON panel through the 1080p JSON array! ([\`50b1c2a\`](https://github.com/acme/foo/commit/50b1c2ae17634194843ca38b0b69993e))
- We need to navigate the haptic SSL program! ([\`b9af129\`](https://github.com/acme/foo/commit/b9af129654524dc7bb7ed8d09684baf8))

### üêõ Fixes

- Use the 1080p HTTP card, then you can input the optical program! ([\`484edcb\`](https://github.com/acme/foo/commit/484edcb1e9154eb3bf861edd35d96c4b))



### üõ† Maintenance

- We need to index the bluetooth PNG pixel! ([\`048bd17\`](https://github.com/acme/foo/commit/048bd17682d84af6a061f2d61c418fde))

### ‚úÖ Tests

- I'll override the digital CSS capacitor, that should bus the EXE alarm! ([\`b04b7d5\`](https://github.com/acme/foo/commit/b04b7d5827d240768b6d2126732cef41))
- You can't generate the system without hacking the multi-byte HDD panel! ([\`d62ac61\`](https://github.com/acme/foo/commit/d62ac612cc36411f9091ee673b9fb265))
- Try to override the COM protocol, maybe it will calculate the solid state capacitor! ([\`0e72523\`](https://github.com/acme/foo/commit/0e72523714e1498dac5639edd805a050))

### üìö Documentation

- navigating the interface won't do anything, we need to transmit the digital THX card! ([\`59ce0fc\`](https://github.com/acme/foo/commit/59ce0fcbdefb4b8487319126bb877a04))
- We need to generate the bluetooth JBOD matrix! ([\`da66679\`](https://github.com/acme/foo/commit/da666790429b4757b60bcd2eb4d05435))
- We need to back up the online PCI interface! ([\`e836705\`](https://github.com/acme/foo/commit/e836705301dc4e93a3c515e5d083d731))

"
`;
